### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by LaskoA**
Please check my solution. I understand that code is a bit bulky, so please advice how to fix it :)

### CODE REVIEW

File: app/parse.py [hunk 1]
```diff
@@ -8,8 +15,68 @@ class Quote:
...
+
+
+def get_num_pages():
+    url = "http://quotes.toscrape.com/page/1/"
+    count = 1
+
+    while True:
+        page = requests.get(url).content
+        soup = BeautifulSoup(page, "html.parser")
+        next_page_element = soup.select_one('li.next > a')
```
Comment (Abnormaltype) [inline app/parse.py:42]: use `""`

File: app/parse.py [hunk 2]
```diff
@@ -8,8 +15,68 @@ class Quote:
...
+
+        if next_page_element:
+            count += 1
+            next_page_url = next_page_element.get('href')
```
Comment (Abnormaltype) [inline app/parse.py:46]: the same

File: app/parse.py [hunk 3]
```diff
@@ -8,8 +15,68 @@ class Quote:
...
+
+def get_single_page_quotes(page_soup):
+    quotes = page_soup.select(".quote")
+    return [parse_single_quote(quote_soup) for quote_soup in quotes]
+
+
+def get_quotes():
+    all_quotes = []
+    num_pages = get_num_pages()
+    url = "http://quotes.toscrape.com/page/"
```
Comment (Abnormaltype) [inline app/parse.py:62]: create a constant

Comment (Abnormaltype) [reply app/parse.py:62]: and DRY

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

**Comment by LaskoA** [comment]
Please check after fixes, thank you.

File: app/parse.py [hunk 4]
```diff
@@ -8,8 +15,67 @@ class Quote:
...
+    ))
+    return Quote(
+        text=single_quote["text"],
+        author=single_quote["author"],
+        tags=single_quote["tags"],
+    )
+
+
+def get_num_pages():
+    url = "http://quotes.toscrape.com/page/1/"
```
Comment (Abnormaltype) [inline app/parse.py:36]: DRY, you have constant URL, use it

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

**Comment by LaskoA** [comment]
Fixed but code works slowlier now..

**Review by Abnormaltype — APPROVED** [review]
Good job!

**Pull request closed without merging**